packages/core/src/services/documents/create.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  DocumentVersion,
+  documentVersions,
+  getDocumentsAtCommit,
+  Result,
+  Transaction,
+} from '@latitude-data/core'
+import { LatitudeError } from '$core/lib/errors'
+
+import { existsAnotherDocumentWithSamePath, getEditableCommit } from './utils'
+
+export async function createNewDocument({
+  commitId,
+  path,
+}: {
+  commitId: number
+  path: string
+}) {
+  const commitResult = await getEditableCommit(commitId)
+  if (commitResult.error) return commitResult
+
+  const currentDocuments = await getDocumentsAtCommit({
+    commitId,
+    commitMergedAt: commitResult.value.mergedAt,
+  })
+  if (currentDocuments.error) return currentDocuments
+
+  if (
+    existsAnotherDocumentWithSamePath({
+      documents: currentDocuments.value,
+      path,
+    })
```

Comment:
hmm we are still doing validations in services.. drizzle is really lacking compared to active record

Reply:
I always thought model validation level was not the right approach. In the end you usually have one or two services that change the entity. I think is better to have in those services the validation.

Reply:
Validations should be closest to the source of truth as possible so to make it difficult to bypass them, this means validations in db when possible, validations closest to db when db validations aren't possible. Services are two steps removed from the db, model just one. Compose your model with validations and you'll be safer.